src/backend/templates/rooms.hbs

```diff
@@ -319,58 +319,54 @@
         $('.fossasia-filter').change(function() {
           var filterVal = $(this).val();
           var temp = JSON.parse(localStorage[eventName]);
-          if (filterVal) {
-            $('.date-filter').each(function() {
-              var flag = 1;
-              $(this).find('.venue-filter').each(function() {
-                var flagVenue = 1;
-                $(this).find('.room-filter').each(function() {
-                  var id = $(this).attr('id');
-                  var elemFlag = 1;
-                  if ($(this).find('.tip-description').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
-                    $(this).find('.session-speakers-list a span').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
-                    $(this).find('.session-speakers-more').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
-                    $(this).find('.event').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0) {
-                    elemFlag = 0;
-                    if(mode && temp[id] != 1) {
-                      elemFlag = 1;
-                    }
-                  }
-                  if(elemFlag == 0) {
-                    flagVenue = flag = 0;
-                    $(this).show();
-                  }
-                  else {
-                    $(this).hide();
+          $('.date-filter').each(function() {
+            var flag = 1;
+            $(this).find('.venue-filter').each(function() {
+              var name = $(this).find('.text').text();
+              var flagVenue = 1;
+              if(roomFilterMode && (name != roomName)) {
+                $(this).hide();
+                return;
+              }
+              $(this).find('.room-filter').each(function() {
+                var id = $(this).attr('id');
+                var elemFlag = 1;
+                if ($(this).find('.tip-description').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
```

Comment:
Can you please write the `chaining` better?
`$(this).find('.tip-description').text().toUpperCase()` to      
                                                                                                    `$(this).`
                                                                                                    ` find('.tip-description').`
                                                                                                     ` text().`
                                                                                                    `  toUpperCase()`